Userland/Libraries/LibWeb/Layout/TreeBuilder.cpp

```diff
@@ -84,8 +84,8 @@ static Layout::Node& insertion_parent_for_block_node(Layout::NodeWithStyle& layo
         return layout_parent;
     }
 
-    // TODO: Floats should probably also be able to have inline siblings
-    if (layout_node.is_absolutely_positioned() && layout_parent.last_child()->children_are_inline()) {
+    if (
+        (layout_node.is_absolutely_positioned() || layout_node.is_floating()) && layout_parent.last_child()->children_are_inline() && layout_parent.last_child()->is_anonymous()) {
```

Comment:
The formatting here looks really weird, let's not have a newline immediately after `if (`